### PR TITLE
Add more Complex methods

### PR DESCRIPTION
**Pull request by tekknolagi**
Also add new specs from upstream ruby/spec.

Continues #608.

### CODE REVIEW

**Comment by tekknolagi** [comment]
It's been a long *long* time since I wrote Ruby, but I think from reading the source this (and potentially other methods) will need to construct an instance from the instance's type, not always base `Complex`. This will make `SomeComplexSubclass.numerator` return an instance of `SomeComplexSubclass` instead of base `Complex`.

I am unfortunately not sure how to call `self.class` from Ruby, so I followed the pattern from the other methods.

**Comment by stevegeek** [comment]
Hi @tekknolagi ! 

Just chiming in here, is the problem here actually that `Complex` it not inheriting from `Numeric` ? 

Eg in Ruby

```
> Complex.ancestors
=> [Complex, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

and Numeric implements `integer?` for example, `https://github.com/ruby/ruby/blob/master/numeric.rb#L31`

While in natalie:

```
$ ./bin/natalie -e "puts Complex.ancestors"
[Complex, Object, Kernel, BasicObject]
```
`Numeric` is not the superclass of Complex. 

Numeric already implements `integer?` here: https://github.com/natalie-lang/natalie/blob/master/src/numeric.rb#L146

**Comment by tekknolagi** [comment]
Sorry, accidentally closed. Great call @stevegeek. I made that change.

**Comment by tekknolagi** [comment]
Hm, that's weird. The spec expects `Complex` not to define `positive?` or `negative?`, which `Numeric` provides. I wonder what upstream does.

**Comment by tekknolagi** [comment]
Oh, looks like they call `rb_undef_method`.

File: src/numeric.rb [hunk 1]
```diff
@@ -184,4 +184,7 @@ def numerator
   def denominator
     to_r.denominator
   end
+
+  undef :positive?
```
Comment (stevegeek) [inline src/numeric.rb:188]: Yeah think thats right, `positive?` & `negative?` are [removed from Complex](https://github.com/ruby/ruby/blob/5ccb625fbbd1e774636a9fdbe0bf1c3d38e085d5/complex.c#L2400) in Ruby, but not from `Numeric`, so need to move these to `Complex`

Comment (tekknolagi) [reply src/numeric.rb:188]: oh, lol, oops. was editing the wrong file

**Review by seven1m — APPROVED** [review]
I admit I don't know a lot about Complex. @stevegeek do you feel good about this PR now?

**Comment by ryangjchandler** [comment]
Complex not inheriting from `Numeric` is my mistake!

**Comment by stevegeek** [comment]
@seven1m @tekknolagi Will take a look bit later and let you know if I notice anything else!

**Comment by seven1m** [comment]
I went ahead and merged. If there's anything we need to change, we can do it in another PR. Thanks all!

**Comment by stevegeek** [comment]
Just one extra comment, seems like the ancestor chain has Comparable twice?

```
TLW:natalie stephen$ ./bin/natalie -e "puts Complex.ancestors"
[Complex, Numeric, Comparable, Comparable, Object, Kernel, BasicObject]
```

Compared to Ruby

```
irb(main):005:0> Complex.ancestors
=> [Complex, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

**Comment by seven1m** [comment]
Whoops! Yeah that's wrong. It might be just reporting it twice though... will you file an issue?

**Comment by stevegeek** [comment]
Done https://github.com/natalie-lang/natalie/issues/705 !

**Pull request merged**